Crash when receiving notifications from other apps while playing a podcast

### Checklist

- [X] I have used the search function for [open](https://github.com/AntennaPod/AntennaPod/issues) **and** [closed](https://github.com/AntennaPod/AntennaPod/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only to occurs with a certain podcast, I will include the URL of that podcast.

### App version

3.0 alpha

### Where did you get the app from

Google Play

### Android version

12

### Device model

_No response_

### First occurred

_No response_

### Steps to reproduce

I am encountering a crash and I have been able to reproduce it consistently. It seems to be caused by receiving notifications from other apps while playing a podcast. See the stack trace below.

### Expected behaviour

_No response_

### Current behaviour

_No response_

### Logs

## Environment
Android version: 12
OS version: 4.9.292-gab4493f31457-ab8272301
AntennaPod version: 3.0.0-beta1
Model: Pixel 3a
Device: sargo
Product: sargo

## Crash info
Time: 07-12-2022 21:30:26
AntennaPod version: 3.0.0-beta1

## StackTrace
```
java.lang.SecurityException: getCallState: Neither user 10339 nor current process has android.permission.READ_PHONE_STATE.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.os.Parcel.readException(Parcel.java:2334)
	at com.android.internal.telecom.ITelecomService$Stub$Proxy.getCallStateUsingPackage(ITelecomService.java:2578)
	at android.telecom.TelecomManager.getCallState(TelecomManager.java:1803)
	at android.telephony.TelephonyManager.getCallState(TelephonyManager.java:5708)
	at de.danoeh.antennapod.core.service.playback.LocalPSMP$1.lambda$onAudioFocusChange$0(SourceFile:870)
	at de.danoeh.antennapod.core.service.playback.LocalPSMP$1.lambda$onAudioFocusChange$0$LocalPSMP$1(Unknown Source:0)
	at de.danoeh.antennapod.core.service.playback.-$$Lambda$LocalPSMP$1$qHsUXc9hJd3By-FdeElAp3T5ZzI.run(Unknown Source:4)
	at de.danoeh.antennapod.core.service.playback.LocalPSMP$PlayerExecutor.submit(SourceFile:94)
	at de.danoeh.antennapod.core.service.playback.LocalPSMP$1.onAudioFocusChange(SourceFile:865)
	at android.media.AudioManager$ServiceEventHandlerDelegate$1.handleMessage(AudioManager.java:3766)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7870)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2195)
	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2223)
	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:907)
	at com.android.server.telecom.TelecomServiceImpl.canReadPhoneState(TelecomServiceImpl.java:2356)
	at com.android.server.telecom.TelecomServiceImpl.access$300(TelecomServiceImpl.java:78)

```

--- a/core/src/main/java/de/danoeh/antennapod/core/service/playback/LocalPSMP.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/playback/LocalPSMP.java
@@ -4,9 +4,10 @@ import android.app.UiModeManager;
 import android.content.Context;
 import android.content.res.Configuration;
 import android.media.AudioManager;
+import android.os.Handler;
+import android.os.Looper;
 import android.os.PowerManager;
 import androidx.annotation.NonNull;
-import android.telephony.TelephonyManager;
 import android.util.Log;
 import android.util.Pair;
 import android.view.SurfaceHolder;
@@ -44,7 +45,6 @@ import de.danoeh.antennapod.playback.base.RewindAfterPauseUtils;
 import de.danoeh.antennapod.core.util.playback.AudioPlayer;
 import de.danoeh.antennapod.core.util.playback.IPlayer;
 import de.danoeh.antennapod.model.playback.Playable;
-import de.danoeh.antennapod.core.util.playback.PlaybackServiceStarter;
 import de.danoeh.antennapod.core.util.playback.VideoPlayer;
 import org.greenrobot.eventbus.EventBus;
 
@@ -66,6 +66,7 @@ public class LocalPSMP extends PlaybackServiceMediaPlayer {
     private volatile boolean pausedBecauseOfTransientAudiofocusLoss;
     private volatile Pair<Integer, Integer> videoSize;
     private final AudioFocusRequestCompat audioFocusRequest;
+    private final Handler audioFocusCanceller;
 
     /**
      * Some asynchronous calls might change the state of the MediaPlayer object. Therefore calls in other threads
@@ -154,6 +155,7 @@ public class LocalPSMP extends PlaybackServiceMediaPlayer {
         this.audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
         this.playerLock = new PlayerLock();
         this.startWhenPrepared = new AtomicBoolean(false);
+        audioFocusCanceller = new Handler(Looper.getMainLooper());
 
         executor = new PlayerExecutor();
         executor.threadPool = new ThreadPoolExecutor(1, 1, 5, TimeUnit.MINUTES, new LinkedBlockingDeque<>(),
@@ -853,53 +855,46 @@ public class LocalPSMP extends PlaybackServiceMediaPlayer {
             if (!PlaybackService.isRunning) {
                 abandonAudioFocus();
                 Log.d(TAG, "onAudioFocusChange: PlaybackService is no longer running");
-                if (focusChange == AudioManager.AUDIOFOCUS_GAIN && pausedBecauseOfTransientAudiofocusLoss) {
-                    pausedBecauseOfTransientAudiofocusLoss = false;
-                    new PlaybackServiceStarter(context, getPlayable())
-                            .callEvenIfRunning(false)
-                            .start();
-                }
                 return;
             }
 
             executor.submit(() -> {
                 playerLock.lock();
-
-                // If there is an incoming call, playback should be paused permanently
-                TelephonyManager tm = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
-                final int callState = (tm != null) ? tm.getCallState() : 0;
-                Log.i(TAG, "Call state:" + callState);
-
-                if (focusChange == AudioManager.AUDIOFOCUS_LOSS ||
-                        (!UserPreferences.shouldResumeAfterCall() && callState != TelephonyManager.CALL_STATE_IDLE)) {
+                if (focusChange == AudioManager.AUDIOFOCUS_LOSS) {
                     Log.d(TAG, "Lost audio focus");
                     pause(true, false);
                     callback.shouldStop();
+                } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK
+                        && !UserPreferences.shouldPauseForFocusLoss()) {
+                    if (playerStatus == PlayerStatus.PLAYING) {
+                        Log.d(TAG, "Lost audio focus temporarily. Ducking...");
+                        setVolumeSync(0.25f, 0.25f);
+                        pausedBecauseOfTransientAudiofocusLoss = false;
+                    }
+                } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT
+                        || focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK) {
+                    if (playerStatus == PlayerStatus.PLAYING) {
+                        Log.d(TAG, "Lost audio focus temporarily. Pausing...");
+                        mediaPlayer.pause(); // Pause without telling the PlaybackService
+                        pausedBecauseOfTransientAudiofocusLoss = true;
+
+                        audioFocusCanceller.removeCallbacksAndMessages(null);
+                        audioFocusCanceller.postDelayed(() -> {
+                            if (pausedBecauseOfTransientAudiofocusLoss) {
+                                // Still did not get back the audio focus. Now actually pause.
+                                pause(true, false);
+                            }
+                        }, 10000);
+                    }
                 } else if (focusChange == AudioManager.AUDIOFOCUS_GAIN) {
                     Log.d(TAG, "Gained audio focus");
+                    audioFocusCanceller.removeCallbacksAndMessages(null);
                     if (pausedBecauseOfTransientAudiofocusLoss) { // we paused => play now
-                        resume();
+                        mediaPlayer.start();
                     } else { // we ducked => raise audio level back
                         setVolumeSync(1.0f, 1.0f);
                     }
-                } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK) {
-                    if (playerStatus == PlayerStatus.PLAYING) {
-                        if (!UserPreferences.shouldPauseForFocusLoss()) {
-                            Log.d(TAG, "Lost audio focus temporarily. Ducking...");
-                            setVolumeSync(0.25f, 0.25f);
-                            pausedBecauseOfTransientAudiofocusLoss = false;
-                        } else {
-                            Log.d(TAG, "Lost audio focus temporarily. Could duck, but won't, pausing...");
-                            pause(false, false);
-                            pausedBecauseOfTransientAudiofocusLoss = true;
-                        }
-                    }
-                } else if (focusChange == AudioManager.AUDIOFOCUS_LOSS_TRANSIENT) {
-                    if (playerStatus == PlayerStatus.PLAYING) {
-                        Log.d(TAG, "Lost audio focus temporarily. Pausing...");
-                        pause(false, false);
-                        pausedBecauseOfTransientAudiofocusLoss = true;
-                    }
+                    pausedBecauseOfTransientAudiofocusLoss = false;
                 }
                 playerLock.unlock();
             });

